core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -325,25 +328,41 @@ private void write(HttpObject o, boolean endOfStream) {
                 // Write an access log if:
                 // - every message has been sent successfully.
                 // - any write operation is failed with a cause.
+                final ChannelFuture failedFuture;
                 if (isSuccess) {
                     maybeLogFirstResponseBytesTransferred();
 
+                    if (state == State.DONE && !endOfStream) {
+                        // if state is DONE, the response should be ended with endOfStream.
+                        return;
+                    }
+
                     if (endOfStream && tryComplete()) {
                         logBuilder().endResponse();
                         reqCtx.log().whenComplete().thenAccept(reqCtx.accessLogWriter()::log);
                     }
 
-                    subscription.request(1);
-                    return;
+                    if (isWritable) {
```

Comment:
I realized that we might need this in `HttpRequestSubscriber` when we send more that one `HttpData`.

Reply:
Good point! Let me look into it. 😀